packages/next/src/lib/interop-default.ts

```diff
@@ -1,3 +1,15 @@
-export function interopDefault(mod: any) {
-  return mod.default || mod
+export function interopDefault<T = any>(mod: unknown): T | undefined {
+  // If the module is falsy (like `null`), just return undefined.
+  if (!mod) return undefined
+
+  // If the module has a default export (named 'default'), return that.
+  if (
+    (typeof mod === 'object' || typeof mod === 'function') &&
+    'default' in mod &&
+    mod.default
+  ) {
+    return mod.default as T
+  }
+
+  return mod as T
```

Comment:
The typing updates here improve the reliability of the return and add optional typescript types to the interopted modules.

I'm open to adjusting this to always return `unknown`, but the `any` return here was too dangerous and resulted in a bug.